AsyncDisplayKit/Layout/ASDimension.h

```diff
@@ -212,6 +212,14 @@ extern ASSizeRange const ASSizeRangeZero;
 extern ASSizeRange const ASSizeRangeUnconstrained;
 
 /**
+ * Returns whether a size range has zero max width or height.
+ */
+ASDISPLAYNODE_INLINE AS_WARN_UNUSED_RESULT BOOL ASSizeRangeIsEmpty(ASSizeRange sizeRange)
+{
+  return (sizeRange.max.width == 0 || sizeRange.max.height == 0);
```

Comment:
Some suggestions on how to make these checks more effective:

- These numbers can be (and have seen them be) less than zero sometimes, due to client code insetting bounds that is often transiently zero sized:
- Let's consider using (value < FLT_EPSILON) as I think this is more formally correct.  This is probably less crucial for 0.0 itself, since floats can represent it exactly.
- One step further than FLT_EPSILON -- since sometimes "dirty", low precision math is used to come up with layout values, I wouldn't be surprised if some apps pass in a value like 0.00272 here occasionally.  I wonder if we should make it (value < 0.1), which while arbitrary, is most likely to help more cases than it hurts (any case that has an intentional bounds of 0.1 points is definitely unusual!)